upf/upf_pfcp_server.c

```diff
@@ -394,6 +396,13 @@ request_t1_expired (u32 seq_no)
 	  && !pool_is_free_index (gtm->nodes, msg->node))
 	{
 	  upf_node_assoc_t *n = pool_elt_at_index (gtm->nodes, msg->node);
+
+	  upf_pfcp_associnfo ("PFCP Association Lost: Node {%s}, Node IP {%U}, Local IP %U, Remote IP %U\n", 
+			      n->node_id.fqdn, 
+			      format_ip46_address, &n->node_id.ip, IP46_TYPE_ANY,
```

Comment:
You can't just assume that IP and FQDN are set in `node_id`. You need to check `node_id.type` and then either print the IP or the FQDN.
Luckily `format_node_id` already does that and can be used here.

Reply:
Yeah you're right, I reckoned it would be more efficient not doing the check and just logging a default value, but that probably wouldn't have worked. Thanks for the heads up on format_node_id. Consider it fixed.